src/REGothEngine.cpp

```diff
@@ -214,60 +226,101 @@ void REGothEngine::shutdown()
   }
 }
 
-int ::REGoth::main(REGothEngine& regoth, int argc, char** argv)
+const EngineConfig& REGothEngine::config() const
 {
-  regoth.initializeBsf();
+  return mConfig;
+}
+
+void ::REGoth::parseArguments(int argc, char** argv, EngineConfig& config)
+{
+  bool help;
+  bool version;
+
+  cxxopts::Options options{argv[0], "REGoth - zEngine Reimplementation."};
+
+  // Add general options.
+  options.add_options()
+    ("h,help", "Print this help message", cxxopts::value<bool>(help))
+    ("version", "Print the REGoth version", cxxopts::value<bool>(version))
+    ("v,verbosity", "Verbosity level", cxxopts::value<bool>())
```

Comment:
I'm afraid this doesn't make sense. I thought that it would be cool if we could increase the verbosity level like in some UNIX programs, e.g. `-vv` is less verbose than `-vvvvv` or so. But it seems that the bsf verbosity can only be changed via the CMake variables :/

Any ideas, or should I just remove that?

Reply:
Check Out the location where `REGOTH_LOG` is defined. We're free to control the verbosity of our logs from there.

Reply:
I think this will take some other thinks to tweak, and this PR is already quite large. I will postpone this to another PR, together with the version output, if that's okay for you?

Reply:
Sure!